tests/MetricExporter.spec.ts

```diff
@@ -42,3 +49,176 @@ describe("MetricExporter", () => {
 		expect(exporter["_reqOpts"].protocol).toEqual("http:");
 	});
 });
+
+describe("MetricExporter.export", () => {
+	beforeEach(() => nock.cleanAll());
+
+	test("should export metrics and return a success message", (done) => {
+		const target_host =  "https://example.com:8080";
+		const target_path = "/metrics";
+		const target_url = target_host + target_path;
+		const exporter = new DynatraceMetricExporter({
+			url: target_url
+		});
+
+		// if this request is not received with a body matching the regex below,
+		// a non-success error code will be returned, making the expect call below
+		// fail.
+		const scope: nock.Scope = nock(target_host)
+			.post(target_path, /test,key=value count,delta=10 \d{13}/g)
+			.once()
+			.reply(200);
+
+		const rec: MetricRecord = getTestMetricRecord("test", 10, { "key": "value" });
+
+		exporter.export([rec],
+			(result: ExportResult) => {
+				expect(result.code).toEqual(ExportResultCode.SUCCESS);
+				// the request was sent once.
+				expect(scope.isDone()).toBe(true);
+				done();
+			});
+	});
+
+	test("should return after normalizing metric name", (done) => {
+		const target_host =  "https://example.com:8080";
+		const target_path = "/metrics";
+		const target_url = target_host + target_path;
+		const exporter = new DynatraceMetricExporter({
+			url: target_url
+		});
+
+		const scope: nock.Scope = nock(target_host)
+			.post(target_path, /_ count,delta=10 /)
+			.once()
+			.reply(200);
+
+		const rec: MetricRecord = getTestMetricRecord("~!@", 10, {});
+
+		exporter.export([rec],
+				(result: ExportResult) => {
+					expect(result.code).toEqual(ExportResultCode.SUCCESS);
+					expect(scope.isDone()).toBe(true);
+					done();
+				});
+	});
+
+	test("should return success on empty list but not send the request", (done) => {
+		const target_host =  "https://example.com:8080";
+		const target_path = "/metrics";
+		const target_url = target_host + target_path;
+		const exporter = new DynatraceMetricExporter({
+			url: target_url
+		});
+
+		const scope: nock.Scope = nock(target_host)
+			.post(target_path)
+			.once()
+			.reply(200);
+
+		exporter.export([],
+				(result: ExportResult) => {
+					expect(result.code).toEqual(ExportResultCode.SUCCESS);
+					expect(scope.isDone()).toBe(false);
```

Comment:
Is this how we assert that no requests should have been sent? I saw `.times()` in the docs but not sure if that should be used instead of `.once()`? Maybe `.times(0)` is what we are after?

Reply:
I figured out that its possible to add a callback on the mock to fire when the mock replies to a request. I just added a `fail` in there, meaning that if the mock ever responds, the test will fail.